membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -121,14 +121,94 @@
+
+    Args:
+        radius (float): Radius of hemisphere
+        nx (int): Number of x gridpoints
+        ny (int): Number of y grid points
+        patch_scale (float, optional): Scale factor for range of grid. Defaults to 5.
+
+    Returns:
+        Tuple[np.ndarray, float, float, float, float]: height field, mean curvature, gaussian curvature, dx, dy
+    """
+    x, dx = np.linspace(-radius / scale, radius / scale, nx, retstep=True)
+    y, dy = np.linspace(-radius / scale, radius / scale, ny, retstep=True)
+    xx, yy = np.meshgrid(x, y, indexing="ij")
+    zz = np.sqrt(radius**2 - xx**2 - yy**2)
+    H = -1 / radius
+    K = 1 / radius**2
+    return zz, H, K, dx, dy
+
+
+def test_parametric_hemisphere_curvature():
+    for radius in np.arange(5, 100, 5):
+        for nx in [7, 23, 47, 67]:
+            for ny in [7, 23, 47, 67]:
+                # Ensure that patch_scale is sufficiently large such that points are closely spaced
+                height_field, H, K, dx, dy = parametric_hemisphere(
+                    radius=radius, nx=nx, ny=ny, scale=200
+                )
+                mean = mean_curvature(height_field, dx, dy)
+                # Check only central points to avoid boundary errors from np.gradient
+                assert np.all(np.isclose(mean[2:-2, 2:-2], H))
+                gauss = gaussian_curvature(height_field, dx, dy)
+                assert np.all(np.isclose(gauss[2:-2, 2:-2], K))
+
+
+def parametric_saddle(
+    length: float, nx: int, ny: int, scale: float = 1000
+) -> tuple[np.ndarray, float, float]:
+    """Generate the height field and analytical mean/gaussian curvature corresponding to a monkey saddle.
+
+    Args:
+        length (float): Sample from - to + length.
+        nx (int): Number of x grid points
+        ny (int): Number of y grid points
+        scale (float, optional): Scale factor for range of grid. Defaults to 500.
+
+    Returns:
+        tuple[np.ndarray, np.ndarray, np.ndarray, float, float]: height field, mean curvature, gaussian curvature, dx, dy
+    """
+    x, dx = np.linspace(-length / scale, length / scale, nx, retstep=True)
+    y, dy = np.linspace(-length / scale, length / scale, ny, retstep=True)
+    xx, yy = np.meshgrid(x, y, indexing="ij")
+    zz = xx * (xx**2 - 3 * yy**2)  # Monkey saddle
+
+    K = -36 * (xx**2 + yy**2) / (1 + 9 * (xx**2 + yy**2) ** 2) ** 2
+    H = (
+        27
+        * xx
+        * (-(xx**4) + 2 * xx**2 * yy**2 + 3 * yy**4)
+        / (1 + 9 * (xx**2 + yy**2) ** 2) ** (3 / 2)
+    )
+    return zz, K, H, dx, dy
+
+
+def test_parametric_saddle_curvature():
+    for nx in [7, 23, 47, 67]:
+        for ny in [7, 23, 47, 67]:
+            height_field, K, H, dx, dy = parametric_saddle(
+                length=10, nx=nx, ny=ny, scale=800
+            )
+            mean = mean_curvature(height_field, dx, dy)
+            gauss = gaussian_curvature(height_field, dx, dy)
+
```

Comment:
Can also check that has 0 gaussian curvature at origin?

Reply:
The 0 gaussian curvature at origin condition is implicitly checked by conditions where nx and ny are odd. If the discretization doesn't include (0, 0) then we'd have to interpolate or skip the test. I think comparison against the analytical formula at corresponding grid points should be robust and complete enough.